lib/data-access/logos.ts

```
@@ -0,0 +1,260 @@
+ * @returns A promise resolving to an object containing the logo buffer, its source, and content type, or `null` if no logo is found.
+ *
+ * @remark
+ * External logo fetching can be skipped via environment variables (`NODE_ENV='test'` or `SKIP_EXTERNAL_LOGO_FETCH='true'`). The function uploads new or changed logos to S3 and caches results in memory.
+ */
+export async function getLogo(domain: string): Promise<{ buffer: Buffer; source: LogoSource; contentType: string } | null> {
+  const override = (globalThis as {getLogo?: typeof getLogo}).getLogo;
+  if (typeof override === 'function' && override !== getLogo) {
+    return override(domain);
+  }
+  const cached = ServerCacheInstance.getLogoFetch(domain);
+  if (cached && cached.buffer) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)